addon/components/model-form.js

```diff
@@ -55,6 +55,10 @@ export default class ModelFormComponent extends Component {
     // not be classes with tracked fields etc. anymore but just pojos. We need to preserve the classes.
     const deepMerge = (original, objectToApply) => {
       Object.entries(objectToApply).forEach(([key, value]) => {
+        if (!value) {
```

Comment:
How will this behave on boolean values?

Reply:
The check should probably look something like this `value !== null && value !== undefined`.

Reply:
Good catch, totally forgot about those